File: api/src/main/java/info/preva1l/sentinel/action/Action.java
package info.preva1l.sentinel.action;

import net.kyori.adventure.key.Key;
import org.jetbrains.annotations.NotNull;

import java.util.UUID;

/**
 * Created on 1/07/2025
 *
 * @author Preva1l
 */
public interface Action {
    @NotNull Key type();

    @NotNull UUID performer();
}

File: common/src/main/kotlin/info/preva1l/sentinel/actions/Action.kt
package info.preva1l.sentinel.actions

import info.preva1l.sentinel.libraries.flavor.service.Configure
import info.preva1l.sentinel.libraries.flavor.service.Service

/**
 * Created on 18/02/2025
 *
 * @author Preva1l
 */
@Service
abstract class Action {
    @Configure
    abstract fun configure()
}